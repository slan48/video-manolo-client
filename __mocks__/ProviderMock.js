import React from 'react';
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import {createWrapper, HYDRATE} from 'next-redux-wrapper';
import {reducer} from "../store";
import {Provider} from "react-redux";

const middleware = [thunk];

// create store
export const store = createStore(reducer, composeWithDevTools(applyMiddleware(...middleware)));

// create a makeStore function
const makeStore = context => store;

// export an assembled wrapper
export const wrapper = createWrapper(makeStore, {debug: process.env.NEXT_PUBLIC_ENV === 'development'});

const ProviderMock = (props) => (
  <Provider store={store}>
    {props.children}
  </Provider>
);

export default wrapper.withRedux(ProviderMock);
