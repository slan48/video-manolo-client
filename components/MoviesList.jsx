import React from 'react';
import {useSelector} from "react-redux";
import Link from "next/link";
import Loader from "./Loader";

const MoviesList = () => {
  const movies = useSelector(state => state.movies)
  return (
    <>
      <div className="movies-list-component">
        {movies && movies.length ?
          <div className="row">
            {movies.map(movie => (
              <div key={movie._id} className="col-md-3">
                <div className="movies-list-item" style={{backgroundImage: `url(${movie.image})`}}>
                  <div className="movie-info-container">
                    <h4>{movie.title}</h4>
                    <div className="movie-info-inner">
                      <p>{movie.description}</p>
                      <Link href="/movies/[movieId]" as={`/movies/${movie._id}`}>
                        <a className="btn btn-outline-light" href={`/movies/${movie._id}`}>
                          Ver m√°s
                        </a>
                      </Link>
                    </div>
                  </div>
                </div>
              </div>
            ))}
          </div>
          :
          <div className="loader-container">
            <Loader />
          </div>
        }
      </div>
      <style jsx>{`
         .movies-list-component{
          padding: 50px 0;          
         }
         
        .movies-list-item{
          height: 300px;
          margin-bottom: 30px;
          position: relative;
          transition: all 0.3s ease;
         
          
          .movie-info-inner{
            text-align: center;
            opacity: 0;
            transition: all 0.5s ease;
            position: absolute;
            padding-bottom: 20px;
          }
          
          &:hover{
            transform: scale(1.1); 
           
            .movie-info-inner{
              position: static;
              opacity: 1;
            }
          }
        }
        
        .movie-info-container{
          position: absolute;
          bottom: 0;
          left: 0;
          height: 50%;
          width: 100%;
          background-image: linear-gradient(180deg,transparent,rgba(0,0,0,.8));
          border-radius: inherit;
          display: flex;
          align-items: center;
          justify-content: flex-end;
          flex-direction: column;
          color:#fff;
        }
      `}</style>
    </>
  );
};

export default MoviesList;
