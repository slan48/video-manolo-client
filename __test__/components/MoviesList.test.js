import React from 'react';
import { mount } from 'enzyme';
import MoviesList from "../../components/MoviesList";
import ProviderMock from "../../__mocks__/ProviderMock";
import { create } from 'react-test-renderer';

describe('<MoviesList />', function () {
  test('MoviesList component render', () => {
    const moviesList = mount(
      <ProviderMock>
        <MoviesList />
      </ProviderMock>
    );
    expect(moviesList.length).toEqual(1);
  })

  test('Loader Snapshot', () => {
    const moviesList = create(
      <ProviderMock>
        <MoviesList />
      </ProviderMock>
    );
    expect(moviesList.toJSON()).toMatchSnapshot();
  })
});
