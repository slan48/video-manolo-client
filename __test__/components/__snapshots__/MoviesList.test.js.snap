// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesList /> Loader Snapshot 1`] = `
<div
  className="jsx-1480517449 movies-list-component"
>
  <div
    className="jsx-1480517449 loader-container"
  >
    <div
      className="jsx-1940039920"
    >
      <div
        className="jsx-1940039920 lds-grid"
      >
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
        <div
          className="jsx-1940039920"
        />
      </div>
    </div>
  </div>
</div>
`;
